src/main/java/works/crossdock/CrossdockClient.java

```diff
@@ -0,0 +1,97 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package works.crossdock;
+
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.handler.codec.http.HttpObjectAggregator;
+import io.netty.handler.codec.http.HttpServerCodec;
+import io.netty.util.concurrent.Future;
+import io.netty.util.concurrent.GenericFutureListener;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import works.crossdock.client.Behavior;
+
+public class CrossdockClient {
+  private final int crossdockPort;
+  private final Map<String, Behavior> behaviors = new HashMap<>();
+
+  /**
+   * Creates a client handle for crossdock tests.
+   *
+   * @param crossdockPort port to run the crossdock server on
+   * @param inputBehaviors map of behaviors and actions to execute
+   */
+  public CrossdockClient(int crossdockPort, Map<String, Behavior> inputBehaviors) {
+    this.crossdockPort = crossdockPort;
+    for (Entry<String, Behavior> entry : inputBehaviors.entrySet()) {
+      behaviors.put(entry.getKey(), entry.getValue());
+    }
+  }
+
+  /**
+   * Starts and stops the crossdock server.
+   *
+   * @throws Exception if the start of server fails
+   */
+  public Future<Void> start() throws Exception {
+    EventLoopGroup bossGroup = new NioEventLoopGroup();
+    EventLoopGroup workerGroup = new NioEventLoopGroup();
+    ServerBootstrap bootstrap = new ServerBootstrap();
+    bootstrap
+        .group(bossGroup, workerGroup)
+        .channel(NioServerSocketChannel.class)
+        .childHandler(
+            new ChannelInitializer<SocketChannel>() {
+              @Override
+              public void initChannel(SocketChannel ch) throws Exception {
+                ch.pipeline()
+                    .addLast(new HttpServerCodec())
+                    .addLast(new HttpObjectAggregator(Integer.MAX_VALUE))
+                    .addLast(new CrossdockServerInboundHandler(behaviors));
+              }
+            })
+        .option(ChannelOption.SO_BACKLOG, 128)
+        .childOption(ChannelOption.SO_KEEPALIVE, true);
+
+    ChannelFuture future = bootstrap.bind("0.0.0.0", crossdockPort).sync();
+    future
+        .channel()
+        .closeFuture()
+        .addListener(
+            new GenericFutureListener<Future<? super Void>>() {
+              @Override
+              public void operationComplete(Future<? super Void> future) throws Exception {
+                workerGroup.shutdownGracefully();
+                bossGroup.shutdownGracefully();
+              }
+            });
+    return future.channel().closeFuture();
```

Comment:
Yes, this is a good suggestion. I've added ChannelFuture as part of the class and exposed a stop method on the client.